agent/inject.js

```diff
@@ -21,9 +21,11 @@ module.exports = function(hook: Hook, agent: Agent) {
       agent.setReactInternals(id, helpers);
       helpers.walkTree(agent.onMounted.bind(agent, id), agent.addRoot.bind(agent, id));
     }),
-    hook.sub('root', ({renderer, internalInstance}) => agent.addRoot(renderer, internalInstance)),
+    hook.sub('rootCommitted', ({renderer, internalInstance}) => agent.rootCommitted(renderer, internalInstance)),
     hook.sub('mount', ({renderer, internalInstance, data}) => agent.onMounted(renderer, internalInstance, data)),
+    hook.sub('root', ({renderer, internalInstance}) => agent.addRoot(renderer, internalInstance)),
     hook.sub('update', ({renderer, internalInstance, data}) => agent.onUpdated(internalInstance, data)),
+    hook.sub('updateProfileTimes', ({renderer, internalInstance, data}) => agent.onUpdatedProfileTimes(internalInstance, data)),
     hook.sub('unmount', ({renderer, internalInstance}) => agent.onUnmounted(internalInstance)),
```

Comment:
Is it worth keeping "basic" events necessary for the tree separate from "profiler" events?

Reply:
Not sure what you're suggesting.

The added "updateProfileTimes" event was necessary to handle the case where a component bailed out– something DevTools traditionally didn't care about, but now needs to handle so that we can update the `actualDuration`

Reply:
Sorry, I proposed to separate 

```js
    // Basic functionality:
    hook.sub('mount', ({renderer, internalInstance, data}) => agent.onMounted(renderer, internalInstance, data)),	    hook.sub('mount', ({renderer, internalInstance, data}) => agent.onMounted(renderer, internalInstance, data)),
    hook.sub('update', ({renderer, internalInstance, data}) => agent.onUpdated(internalInstance, data)),	    hook.sub('update', ({renderer, internalInstance, data}) => agent.onUpdated(internalInstance, data)),
    hook.sub('unmount', ({renderer, internalInstance}) => agent.onUnmounted(internalInstance)),	    hook.sub('unmount', ({renderer, internalInstance}) => agent.onUnmounted(internalInstance)),

    // Necessary for profiler:
    hook.sub('root', ({renderer, internalInstance}) => agent.addRoot(renderer, internalInstance)),
    hook.sub('rootCommitted', ({renderer, internalInstance}) => agent.rootCommitted(renderer, internalInstance)),
    hook.sub('updateProfileTimes', ({renderer, internalInstance, data}) => agent.onUpdatedProfileTimes(internalInstance, data)),
```

So it's clearer the new set of events are additional and aren't necessary for the main part to work.

Reply:
Ah! Yeah that makes sense 😄 I misunderstood.